File: 1897-redistribute-characters-to-make-all-strings-equal/1897-redistribute-characters-to-make-all-strings-equal.cpp
class Solution {
public:
    bool makeEqual(vector<string>& words) {
        int freq[26]={0};
        
        for(string word: words)
        {
            for(char c : word)
            {
                freq[c-'a']++;
            }
        }
        for(int i=0; i<26; i++)
        {
            if(freq[i]%words.size()!=0)
            {
                return false;
            }
        }
        return true;
    }
};

File: 1897-redistribute-characters-to-make-all-strings-equal/1897-redistribute-characters-to-make-all-strings-equal.java
class Solution {
    public boolean makeEqual(String[] words) {
        int n=words.length;
        
        HashMap<Character, Integer> hm= new HashMap<>();
        
        for(String word: words)
        {
            for(char c: word.toCharArray())
            {
                hm.put(c,hm.getOrDefault(c,0)+1);
            }
        }
        for(char key: hm.keySet())
        {
            if(hm.get(key)%n!=0)
            {
                return false;
            }
        }
        return true;
    }
}